ledger/store/trackerdb/generickv/accounts_ext_writer.go

```diff
@@ -0,0 +1,220 @@
+	// if catchpointStaging {
+	// 	id = "catchpointStaging"
+	// }
+	// "REPLACE INTO accounttotals
+	//		(id, online, onlinerewardunits, offline, offlinerewardunits, notparticipating, notparticipatingrewardunits, rewardslevel)
+	// VALUES (?, ?, ?, ?, ?, ?, ?, ?)"
+
+	// write totals entry
+	raw := protocol.Encode(&totals)
+	err = w.kvw.Set(totalsKey(catchpointStaging), raw)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (w *accountsWriter) OnlineAccountsDelete(forgetBefore basics.Round) (err error) {
+	// The SQL at the time of impl:
+	//
+	// SELECT
+	//		rowid, address, updRound, data
+	// FROM onlineaccounts
+	// WHERE updRound < ?
+	// ORDER BY address, updRound DESC
+	//
+	// The it would delete by  rowid in chunks with:
+	//
+	// 		DELETE FROM onlineaccounts WHERE rowid IN (..)
+
+	// On the KV implmentation:
+	//
+	// We have two ranges of keys associated with online accounts:
+	// - the `onlineAccountKey(address, round)` -> "-".join(kvPrefixOnlineAccount, addr, round)
+	// - and the `onlineAccountBalanceKey(round, normBalance, addr) -> "-".join(kvPrefixOnlineAccountBalance, round, normBalance, addr)
+
+	// 1. read from the `onlineAccountBalanceKey` range since we can the addr's that will need to be deleted
+	start := []byte(kvPrefixOnlineAccountBalance + "-")
+	end := []byte(kvPrefixOnlineAccountBalance + ".")
+	iter := w.kvr.NewIter(start, end, false)
+	defer iter.Close()
+
+	toDelete := make([]struct {
+		basics.Address
+		basics.Round
+	}, 0)
+
+	for iter.Next() {
+		// read the key
+		// schema: <prefix>-<rnd>-<balance>-<addr>
+		key := iter.Key()
+
+		// extract the round from the key (offset: 1)
+		rndOffset := len(kvPrefixOnlineAccountBalance) + 1
+		u64Rnd := binary.BigEndian.Uint64(key[rndOffset : rndOffset+8])
+		round := basics.Round(u64Rnd)
+
+		// get the offset where the address starts
+		addrOffset := len(kvPrefixOnlineAccountBalance) + 1 + 8 + 1 + 8 + 1
+		var addr basics.Address
+		copy(addr[:], key[addrOffset:addrOffset+32])
+
+		// mark the item for deletion
+		toDelete = append(toDelete, struct {
+			basics.Address
+			basics.Round
+		}{addr, round})
+	}
+
+	// 2. delete the individual addr+round entries
+	for _, item := range toDelete {
+		// TODO: [perf] we might be able to optimize this with a SingleDelete call
+		err = w.kvw.Delete(onlineAccountKey(item.Address, item.Round))
+		if err != nil {
+			return
+		}
+	}
+
+	// 3. delete the range from `onlineAccountBalanceKey`
+	err = w.kvw.DeleteRange(start, end)
+	if err != nil {
+		return err
+	}
+
+	return
+}
+
+func (w *accountsWriter) AccountsPutOnlineRoundParams(onlineRoundParamsData []ledgercore.OnlineRoundParamsData, startRound basics.Round) error {
+	// The SQL at the time of impl:
+	//
+	// for i, data := range onlineRoundParamsData {
+	// 		the inserted rnd value is startRound + i
+	//
+	//		INSERT INTO onlineroundparamstail (rnd, data) VALUES (?, ?)
+	//
+
+	// insert the round params
+	for i, params := range onlineRoundParamsData {
```

Comment:
Don't used `params` here, just index it directly in the loop. It will be safer and will avoid the extra copy in each iteration.